electron-1/preload.js

```diff
@@ -0,0 +1,25 @@
+const { ipcRenderer } = require('electron');
+
+var BX = function () {
+
+    return {
+        getName: () => {
+            return new Promise((resolve) => {
+                ipcRenderer.send('name', 'Hello');
+                ipcRenderer.on('nameAnswer', (event, name) => {
+                    resolve(name);
+                });
+            });
+        },
+        getVersion: () => {
+            return new Promise((resolve) => {
+                ipcRenderer.send('version', 'Hello');
```

Comment:
Have you considered `ipcRenderer.sendSync`? Why it's not a good option?

Reply:
I thought about it, but it looks like it's not recommended as it block the renderer process. The impact would have been I think invisible in this case, as the `app.getName()` and `app.getVersion()` call must be very quick, but it is not good practice.